### DISCUSSION TITLE
What's the point of having a lot of this kind of writing in the code?

### DISCUSSION THREAD

**Discussion by junmaqiang** [fyne-io/fyne#3419] [Q&A] [answered]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

`
var _ CanvasObject = (*Container)(nil)
`

`
var _ Dialog = (*dialog)(nil)
`

why write like this？

### Is it possible to construct a solution with the existing API?

why write like this

### Describe the solution you'd like to see.

why write like this

**Comment by andydotxyz** [comment]
This seems like a general question, probably better targetted as a discussion, converting

**Comment by andydotxyz** [comment, marked as answer]
It is to declare that a type conforms to an interface - so you get compile errors if the object fails to implement a method of the interface.
Normally in Fyne code you will see a comment with it like:
```go
// Declare conformity with CanvasObject interface
```

**Comment by junmaqiang** [comment]
thank you, I Understood it now
